quickwit/quickwit-metastore/src/metastore/postgres/metastore.rs

```diff
@@ -399,6 +399,41 @@ impl MetastoreService for PostgresqlMetastore {
         Ok(response)
     }
 
+    async fn update_index(
+        &mut self,
+        request: UpdateIndexRequest,
+    ) -> MetastoreResult<IndexMetadataResponse> {
+        let retention_policy_opt = request.deserialize_retention_policy()?;
+        let search_settings = request.deserialize_search_settings()?;
+        let index_uid: IndexUid = request.index_uid().clone();
+        let mut mutated_metadata_opt = None;
+        let mutated_metadata_ref = &mut mutated_metadata_opt;
+        run_with_tx!(self.connection_pool, tx, {
+            mutate_index_metadata::<MetastoreError, _>(
```

Comment:
Time to refactor `mutate_index_metadata` to return `MetastoreResult<MutationOccurred<T>, E>` instead of `MetastoreResult<bool, E>`.